### ISSUE TITLE
[ChromatiCraft] Locks and Keys core will generate in the wrong direction on easiest difficulty

### ISSUE THREAD

**Issue by thehajo** [ReikaKalseki/Reika_Mods_Issues#2675]
When i first did Locks and Keys, i was very disappointed since i assumed the core failed to generate. But after my friend encountered the same, i said jokingly to just veinmine around to see if you maybe just entered some numbers wrong.
Well, seems like there is a "-" missing, or too much, as it just generated, a fair bit apart, on the opposite end. To recreate this and submit this, i used WorldEdit to clear the stone.
https://imgur.com/a/LmoW4jV
I hope it is not too small and still visible.
This is on a server on V26b, and "Dimension Structure Difficulty"=1.

**Comment by ReikaKalseki** [comment]
Does this happen reliably or only occasionally?

**Comment by thehajo** [comment]
Happened in every puzzle we did so far, which was i think 4. First one i gave up since i didn't knew it. Second and third one, my friend and me got the cores, and 4th one was to get the screenshots.

**Comment by ReikaKalseki** [comment]
Looking at the code, it looks like the room generates in the same direction no matter what - likely a legacy of when the puzzle was linearly arranged - as opposed to its current back and forth - and as such is misaligned to your ending.

What puzzle difficulty are you on? Does this change the generation?

**Comment by thehajo** [comment]
Like i said, we play on "Dimension Structure Difficulty"=1. Have not yet tried it with other difficulties. Can do and will report back later.

**Comment by thehajo** [comment]
So, after testing it only happens on difficulty 1. I tried it with 2 and 3 and it was fine there, don't know if 0 and 4 are a thing. 

I used the chromaprog command to maximze my progress in a singleplayer world, and while trying to find the Locks and Keys puzzle, i noticed some differences between the actual colors and the puzzle that correlates with it, and the puzzles noted down for each color in the book. I hope it is clear what i mean. I only noted it down for difficulty 1.
https://imgur.com/a/lvlQwCm
Kurauri was Locks and Keys for difficulty 1 and difficulty 2, however when it came to difficulty 3, it was now Crystal Music. However the book seemed to be correct then, and i was able to easily find Locks and Keys on difficulty 3 at Portokali.
